GVRf/Framework/framework/src/main/java/org/gearvrf/x3d/X3DparseLights.java

```diff
@@ -0,0 +1,398 @@
+
+	
+    public X3DparseLights(GVRContext gvrContext, GVRModelSceneObject root) {
+    	this.gvrContext = gvrContext;
+    	//this.activityContext = gvrContext.getContext();
+    	//this.assetManager = activityContext.getAssets();
+    	this.root = root;
+    }
+
+
+	public void Parse(InputStream inputStream, ShaderSettings shaderSettings) {
+	      try {
+		      //System.out.println("X3DparseLights BEGIN");
+	    	  this.shaderSettings = shaderSettings;
+	          SAXParserFactory factory = SAXParserFactory.newInstance();
+	          SAXParser saxParser = factory.newSAXParser();
+	          UserHandler userhandler = new UserHandler();
+	          saxParser.parse(inputStream, userhandler);
+		      //System.out.println("X3DparseLights END");
+	       } catch (Exception e) {
+	          e.printStackTrace();
+	       }
+	       
+	}  // end Parse
+
+	
+	class UserHandler extends DefaultHandler {
+
+		   String attributeValue = null;
+
+		   public float[] parseFixedLengthFloatString(String numberString, int componentCount, boolean constrained0to1, boolean zeroOrGreater) {
+			   StringReader sr = new StringReader(numberString);
+			   StreamTokenizer st = new StreamTokenizer(sr);
+			   st.parseNumbers();
+			   int tokenType;
+			   float componentFloat[] = new float[componentCount];
+			   try {
+				   for (int i = 0; i < componentCount; i++) {
+					   if ((tokenType = st.nextToken()) == StreamTokenizer.TT_NUMBER) {
+						   componentFloat[i] = (float) st.nval;
+						   if (constrained0to1) {
+							   if (componentFloat[i] < 0) componentFloat[i] = 0;
+							   else if  (componentFloat[i] > 1) componentFloat[i] = 1;
+						   }
+						   else if (zeroOrGreater) {
+							   if (componentFloat[i] < 0) componentFloat[i] = 0;
+						   }
+					   }
+				   }
+			   }
+			   catch (IOException e) {
+				   Log.d("Parse X3D", "Error: " + e);
+			   }
+			   return componentFloat;
+		   }
+
+		   public boolean parseBooleanString(String booleanString) {
+			   StringReader sr = new StringReader(booleanString);
+			   StreamTokenizer st = new StreamTokenizer(sr);
+			   boolean value = false;
+			   int tokenType;
+				try {
+					tokenType = st.nextToken();
+					if ( tokenType == StreamTokenizer.TT_WORD) {
+						if ( st.sval.equalsIgnoreCase("true")) value = true;
+					}
+				} catch (IOException e) {
+					   Log.d("Parse X3D", "Boolean Error: " + e);
+					e.printStackTrace();
+				}
+			   return value;
+		   }
+		   
+		   
+	/*********** Parse the X3D File **************/
+	   @Override
+	   public void startElement(String uri,
+	      String localName, String qName, Attributes attributes)
+	         throws SAXException {
+
+		      /********** Transform **********/
+		      if (qName.equalsIgnoreCase("transform")) {
+		        System.out.print("<" + qName + " ");
+		        attributeValue = attributes.getValue("DEF");
+		        if (attributeValue != null) {
+		        	System.out.print("DEF='" + attributeValue + "' ");
```

Comment:
Thanks - left over code from debugging the parser.  Removed in the next version or replaced with Log call.